### ISSUE TITLE
[regression] shrinking function heads to 1-line moving comments between if stanzas

### ISSUE THREAD

**Issue by novaugust** [adobe/elixir-styler#67]
I have a very large file where a new comment was being moved when nothing else was. The file's so big I'm unsure how to create a minimal repro at this point, but the statements were fairly deeply nested within a function

```diff
defmodule ...
  def fun
    if a do
      another_fun = fn ->
        with ...
           # see what i mean? decent bit of nesting

           if foo do
             bar
+
+            # a
+            # long
+            # comment
+            # moved :/
           end

-          # a
-          # long
-          # comment
-          # moved :/
           if widget != wodget do
```

**Comment by novaugust** [comment]
the function head in question is split over multiple lines, so this is likely a regression in function-head shrinking

**Comment by novaugust** [comment]
i have a theory that this occurs when we try to shrink the function head, but ultimately the formatter disallows the shrink - so it's like

1. we shrink the function head
2. we displace the comments an appropriate amount
3. formatter expands the function head again, as it was too long
4. however, the comments were still moved up

soooo, a problem resulting from our lazy shrinks? only fix in that case would be to do the heavy lifting of making the document algebra for each head and attempting to shrink it. at that point, might be easier to see if we can modify things upstream in formatter itself to support this collapsing 🤔

**Comment by APB9785** [comment]
Also reproduceable with `case`:

```elixir
def my_function(
      very_long_name,
      so_long_that_this_head_will_not_fit_on_one_line
    ) do
  result =
    case foo do
      :bar -> :baz
      :baz -> :bong
    end

  # My comment
  Context.process(result)
end
```
will become

```elixir
def my_function(
      very_long_name,
      so_long_that_this_head_will_not_fit_on_one_line
    ) do
  result =
    case foo do
      :bar ->
        :baz

      :baz ->
        :bong

        # My comment
    end

  Context.process(result)
end
```

**Comment by novaugust** [comment]
wow thanks andrew. for some reason your simple case made me have an epiphany around what's actually going wrong here. i've got a fix coming =)

**Comment by novaugust** [comment]
fixed in 0.10.1
